x-pack/plugins/apm/public/components/shared/DiscoverButtons/DiscoverButton.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import React from 'react';
+import { StringMap } from 'x-pack/plugins/apm/typings/common';
+import {
+  getAPMIndexPattern,
+  ISavedObject
+} from '../../../services/rest/savedObjects';
+import { KibanaLink } from '../../../utils/url';
+
+interface Props {
+  query: StringMap;
+}
+
+interface State {
+  indexPattern?: ISavedObject;
+}
+
+export class DiscoverButton extends React.Component<Props, State> {
+  public state = {};
+  public async componentDidMount() {
+    const indexPattern = await getAPMIndexPattern();
+    this.setState({ indexPattern });
+  }
+
+  public render() {
+    const { query, children, ...rest } = this.props;
+    const id: string = get(this.state, 'indexPattern.id');
```

Comment:
If we do `this.state.indexPattern && this.state.indexPattern.id` will TS infer the type from the ISavedObject type?

Reply:
if I do `const id = this.state.indexPattern && this.state.indexPattern.id;` I get:
```
[ts] Object is possibly 'undefined'. [2532]
```

Doesn't make any sense since code analysis should infer the guard...

Reply:
Okay, pushed a fix for this so I was now able to drop `_.get`